### PR TITLE
[r/ci] Unbreak `pkgdown` CI

### PR DESCRIPTION
**Pull request by johnkerl**
Addresses #2702. See also particular CI run https://github.com/single-cell-data/TileDB-SOMA/actions/runs/9446113943

### CODE REVIEW

**Comment by johnkerl** [comment]
> Can you carry the same fix over to the nightly valgrind run? Thanks!

Sure thing, once I've got pkgdown unbroken. Just got back from AFK & am now 👀

**Comment by johnkerl** [comment]
Nightly run for `kerl/unbreak-pkgdown-ci` branch (note the `pkgdown` CI normally runs only nightly so it needs a manual trigger):
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/9453591532

**Comment by johnkerl** [comment]
Manual workflow run for `pkgdown` CI (which is normally nightly):
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/9467435693

**Comment by johnkerl** [comment]
@mojaveazure the manual run ☝️ of `pkgdown` CI on this PR is green -- I'll make a separate PR for `r-valgrind` CI

**Review by mojaveazure — APPROVED** [review]
Looks great, thanks for taking care of this!

File: apis/r/tools/install-pkgdown-dependencies.sh [hunk 1]
```diff
@@ -5,5 +5,7 @@ export R_REMOTES_NO_ERRORS_FROM_WARNINGS='true'
 Rscript -e 'install.packages(c("remotes", "pkgdown"))'
 cd apis/r
 Rscript -e 'remotes::install_deps(dependencies = TRUE, upgrade=FALSE)'
-Rscript -e 'remotes::install_local()'
+# Need dependencies = FALSE to avoid getting latest/sometimes-too-new r-tiledb from CRAN when we
+# really want it from our r-universe where we control the version we want:
+Rscript -e 'remotes::install_local(dependencies = FALSE)'
```
Comment (eddelbuettel) [inline apis/r/tools/install-pkgdown-dependencies.sh:10]: Just for completeness:  This is the equivalent base R step:  

```r
install.packages(".", repos=NULL)
```

where repos=NULL is used to install from download tar.gz or the source directory you may be in.

We already have `remotes` installed and need it so no need to change.

Comment (johnkerl) [reply apis/r/tools/install-pkgdown-dependencies.sh]: https://github.com/single-cell-data/TileDB-SOMA/pull/2703/commits/3ae9824a28e713752f5ddc8d917c77631a3de8df

**Pull request merged**